build: update `pyproject.toml` to follow PEP 639

walid83-0 opened the pull request:
Hi, I updated pyproject.toml following PEP 639.
Please review and merge if everything is fine.

walid83-0 committed:
update pyproject.toml to follow PEP 639

walid83-0 committed:
Merge branch 'main' into main

chalmerlowe committed:
Merge branch 'main' into main

chalmerlowe commented:
@walid83-0 

I am curious about the decision to provide a minimum version number for setuptools.  What is the rationale for this decision? And why such a recent version of setuptools (i.e. "77.0.3"), which, if memory serves, came out in the past six months.

As a general rule, Google tends to avoid putting too many limits on dependency versions as our library users are a global audience with many different environments. We want to be able to enable customers to use this library with as little friction as possible.

For projects in the Googleapis organization that have `pyproject.toml` files, I can only find three that include a version number with setuptools and those versions are pretty old `{"42", "61.0"}`

ACTION: unless we have a really good reason, I request that the version bump be stripped out entirely.

walid83-0 commented:
Thanks for pointing this out. I mistakenly pinned setuptools to 77.0.3 simply because it was the most recent release. I understand now that this creates unnecessary friction for users. Since we don’t depend on any features from that specific version, I’ll update the requirement to be more flexible (>=61) or remove it entirely.
I’m still learning and just wanted to contribute, so thanks for the guidance.

chalmerlowe commented:
Let's just remove the version number completely, please. Thanks!

walid83-0 committed:
Merge branch 'main' into main

walid83-0 commented:
/

walid83-0 committed:
Update pyproject.toml PEP 639

Thanks for the feedback,
I've removed the version number completely as requested.

chalmerlowe commented on pyproject.toml:
```diff
@@ -19,7 +19,8 @@ build-backend = "setuptools.build_meta"
 name = "google-cloud-bigquery"
 authors = [{ name = "Google LLC", email = "googleapis-packages@google.com" }]
-license = { text = "Apache 2.0" }
+license = { text = "Apache-2.0" }
```
Dicts appear to be deprecated. It appears that the correct format is a simple string:
Fixing this should clear up the failing tests. <fingers crossed>.

```suggestion
license = "Apache-2.0"
```

chalmerlowe replied:
@walid83-0 Take a look. The tests are failing. Here is an example Error:

```
× Getting requirements to build editable did not run successfully.
  │ exit code: 1
  ╰─> [44 lines of output]
      configuration error: `project.license` must be string
      GIVEN VALUE:
          {
              "text": "Apache-2.0"
          }
      
      OFFENDING RULE: 'type'
      
      DEFINITION:
          {
              "type": "string"
          }
      Traceback (most recent call last):
```

chalmerlowe committed:
Update pyproject.toml

chalmerlowe committed:
Merge branch 'main' into main
